src/hoc/useNodes.tsx

```diff
@@ -0,0 +1,16 @@
+import { useSelector as reduxUseSelector, useDispatch, TypedUseSelectorHook } from 'react-redux';
+import { addNode, AddNode } from '@/module/node/actions';
+import { IGraphState, INodesState } from '@/type/types';
+
+const useSelector: TypedUseSelectorHook<IGraphState> = reduxUseSelector;
```

Comment:
I don't understand what is happening here? 

useSelector is imported under a different alias, then we declare a variable with the same name that is the same as the imported useSelector, and then we pass a select function to it. 

Can't we just use the useSelector out of the box?